http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/sam/PetNotebookServiceAccountProvider.scala

```diff
@@ -1,45 +1,45 @@
 package org.broadinstitute.dsde.workbench.leonardo.auth.sam
 
-import com.typesafe.config.Config
+import cats.Monad
+import cats.implicits._
+import cats.mtl.ApplicativeAsk
+import org.broadinstitute.dsde.workbench.leonardo.dao.SamDAO
 import org.broadinstitute.dsde.workbench.leonardo.model.ServiceAccountProvider
-import org.broadinstitute.dsde.workbench.model.{UserInfo, WorkbenchEmail}
 import org.broadinstitute.dsde.workbench.model.google.GoogleProject
-
-import scala.concurrent._
+import org.broadinstitute.dsde.workbench.model.{TraceId, UserInfo, WorkbenchEmail}
+import org.http4s.headers.Authorization
+import org.http4s.{AuthScheme, Credentials}
 
 /**
-  * Created by rtitle on 12/5/17.
-  *
   * This is not the default ServiceAccountProvider.
   *
   * To enable, change the configuration value serviceAccounts.providerClass to
   * org.broadinstitute.dsde.workbench.leonardo.auth.PetNotebookServiceAccountProvider
   *
   * See the README for more information.
   */
-class PetNotebookServiceAccountProvider(val config: Config) extends ServiceAccountProvider(config) with SamProvider {
+class PetNotebookServiceAccountProvider[F[_]: Monad](samDAO2: SamDAO[F]) extends ServiceAccountProvider[F] {
```

Comment:
We may want to just delete this class.. it's not used in prod and if we _were_ to use it I think we would have a security vulnerability (I can give more details offline if you're curious :)).